dali/operators/reader/loader/numpy_loader_gpu.cc

```diff
@@ -105,12 +105,12 @@ void NumpyLoaderGPU::ReadSample(ImageFileWrapperGPU& imfile) {
     imfile.transpose_fortan_order = target.fortran_order;
   };
 
-  imfile.read_sample_f = [this, image_file, &imfile] (void *buffer, Index offset,
-                                                      size_t total_size) {
+  imfile.read_sample_f = [this, image_file, &imfile] (void *buffer, Index file_offset,
+                                                      size_t read_size) {
     // read sample
-    ReadSampleHelper(imfile.file_stream.get(), imfile, buffer, offset, total_size);
+    ReadSampleHelper(imfile.file_stream.get(), imfile, buffer, file_offset, read_size);
     // close the file handle
-    imfile.file_stream->Close();
+    // imfile.file_stream->Close();
```

Comment:
When the file is closed now?

Reply:
It is closed in:

dali/operators/reader/numpy_reader_gpu_op.cc 

line 89ff

Reply:
I know it is somehow not great to split it that way, so maybe there is some better, callback based method. I leave that to you to improve design, this is more of a functional demo.